sdk/src/main/java/com/deploygate/sdk/DeployGate.java

```diff
@@ -139,16 +143,43 @@ public void onEvent(
                 Logger.d("collect-device-status event received: %s", targetUri);
 
                 ContentValues cv = new ContentValues();
-                if (sBuildEnvironment != null && !sBuildEnvironment.isEmpty()) {
-                    cv.put(DeployGateEvent.ATTRIBUTE_KEY_BUILD_ENVIRONMENT, sBuildEnvironment.toJsonString());
+                synchronized (getBuildEnvironment().attributes) {
+                    if (!getBuildEnvironment().attributes.isEmpty()) {
+                        String buildEnvironmentJson = new JSONObject(getBuildEnvironment().attributes).toString();
+                        cv.put(DeployGateEvent.ATTRIBUTE_KEY_BUILD_ENVIRONMENT, buildEnvironmentJson);
+                    }
+                }
+
+                ConcurrentHashMap<String, Object> mergedRuntimeExtraHashMap = new ConcurrentHashMap<>();
+                // attribute keys of runtime extra created by user need to prefix with the package name
+                synchronized (getRuntimeExtra().attributes) {
+                    if (!getRuntimeExtra().attributes.isEmpty()) {
+                        for (Map.Entry<String, Object> attr : getRuntimeExtra().attributes.entrySet()) {
+                            String userKey = String.format("%s.%s", mHostApp.packageName, attr.getKey());
```

Comment:
We need to append prefix to attribute keys created by user.

Reply:
Please don't add the extra time-consuming processing. If we append the prefix in CustomAttributes, doesn't it work?